### PR TITLE
refactor(theme-classic): replace color mode toggle with button; remove switchConfig

### PR DESCRIPTION
**Pull request by Josh-Cena**
## Breaking change

- `Toggle` component is renamed to `ColorModeToggle`. Please update your swizzled components.
- `colorMode.switchConfig` is deprecated. If you want to customize the icons for light and dark mode, swizzle `IconLightMode`, `IconDarkMode`, or `ColorModeToggle` instead.

## Motivation

Part of #6717 

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

**Before**

image

image

**After**

image

image

### CODE REVIEW

File: packages/docusaurus-theme-classic/src/theme/Toggle/index.tsx [hunk 1]
```diff
@@ -5,113 +5,88 @@
...
-    onChange,
-  }: Props & {
-    switchConfig: ColorModeConfig['switchConfig'];
-    disabled: boolean;
-  }): JSX.Element => {
-    const {darkIcon, darkIconStyle, lightIcon, lightIconStyle} = switchConfig;
-    const [checked, setChecked] = useState(defaultChecked);
-    const [focused, setFocused] = useState(false);
-    const inputRef = useRef<HTMLInputElement>(null);
+function Toggle({
```
Comment (Josh-Cena) [inline packages/docusaurus-theme-classic/src/theme/Toggle/index.tsx:18]: Review with hide white space

**Comment by slorber** [comment]
That looks nice

Was wondering what's your opinion on having some kind of SVG interpolation morphing one svg to another?

https://web.dev/building-a-theme-switch-component/

Also like this one: https://www.joshwcomeau.com/

**Comment by Josh-Cena** [comment]
Seems cool. Would be a useful extension

**Review by slorber — APPROVED** [review]
LGTM 👍 

As this is a breaking change, I'm going to also rename this one to `ColorModeToggle`

File: packages/docusaurus-theme-classic/src/getSwizzleConfig.ts [hunk 2]
```diff
@@ -7,94 +7,116 @@
 
 import type {SwizzleConfig} from '@docusaurus/types';
 
+/* eslint sort-keys: "error" */
```
Comment (Josh-Cena) [inline packages/docusaurus-theme-classic/src/getSwizzleConfig.ts:10]: 👍

**Pull request merged**